src/linpde_gp/linfunctls/_stacked.py

```diff
@@ -0,0 +1,30 @@
+import functools
+import numpy as np
+
+from ._linfunctl import LinearFunctional
+from ._flattened import FlattenedLinearFunctional
+
+class StackedLinearFunctional(LinearFunctional):
+    def __init__(self, L1: LinearFunctional, L2: LinearFunctional):
+        assert L1.input_shapes == L2.input_shapes
+        self._L1 = FlattenedLinearFunctional(L1)
+        self._L2 = FlattenedLinearFunctional(L2)
+
+        output_shape = (self._L1.output_shape[0] + self._L2.output_shape[0],)
+
+        super().__init__(
+            input_shapes=self._L1.input_shapes,
+            output_shape=output_shape
+        )
+
+    @property
+    def L1(self) -> FlattenedLinearFunctional:
+        return self._L1
+    
+    @property
+    def L2(self) -> FlattenedLinearFunctional:
+        return self._L2
+
+    @functools.singledispatchmethod
+    def __call__(self, f, /, **kwargs):
+        return np.concatenate((self._L1(f), self._L2(f)))
```

Comment:
I'm not sure. Maybe we should discuss the purpose of `StackedLinearFunctional`?
My original idea was that it is simply a utility to combine different linear functionals - even if they potentially have different output shapes. So even linear functionals with shapes (20, 40) and (10, 50, 80) would be stackable. To make this work internally, I would wrap those linear functionals with a `ReshapedLinearFunctional` to make their outputs flat. In this case, I don't see the advantage of using `np.stack`, adding a new dummy dimension, over using `np.concatenate`, which keeps everything flat.

But I'm not sure if it's even that easy, especially once we start integrating KeOps. What do you think?

Reply:
Ah, I see. In this case I would suggest we use the name `ConcatenatedLinearFunctional` and stick to the `np.concatenate` internally, since there might be cases where we actually need a `StackedLinearFunctional`.